labgrid/driver/resetdriver.py

```diff
@@ -24,3 +27,61 @@ def reset(self):
         self.output.set(True)
         time.sleep(self.delay)
         self.output.set(False)
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class BCUResetDriver(Driver, ResetProtocol, BootCfgProtocol):
+    """BCUResetDriver - Driver using https://github.com/NXPmicro/bcu and board
+    serial port as USBResetPort to reset the target into different states"""
+
+    bindings = {
+       "port": {USBResetPort, NetworkUSBResetPort}, 
+    }
+
+    delay = attr.ib(default=1000, validator=attr.validators.instance_of(int))
+    emmc_cfg = attr.ib(default="emmc", validator=attr.validators.instance_of(str))
+    sd_cfg = attr.ib(default="sd", validator=attr.validators.instance_of(str))
+    usb_cfg = attr.ib(default="usb", validator=attr.validators.instance_of(str))
+    qspi_cfg = attr.ib(default="", validator=attr.validators.instance_of(str))
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        self.tool = 'bcu'
+
+    @Driver.check_active
+    @step()
+    def reset(self, mode=""):
+        cmd = self.port.command_prefix + [
+            self.tool,
+            "reset", mode,
+            "-board={}".format(self.port.board),
+            "-id={}".format(self.port.path),
+            "-delay={}".format(str(self.delay)),
+        ]
+        try:
+            print(str(processwrapper.check_output(cmd).decode()))
+        except subprocess.CalledProcessError as e:
+            raise  
+            
+    @Driver.check_active
+    @step()
+    def sd(self):
+        self.reset(self.sd_cfg)
+
+    @Driver.check_active
+    @step()
+    def emmc(self):
+        self.reset(self.emmc_cfg)
+
+    @Driver.check_active
+    @step()
+    def usb(self):
+        self.reset(self.usb_cfg)
+
+    @Driver.check_active
+    @step()
+    def qspi(self):
+        if not self.qspi_cfg:
+            print("QSPI mode is not set, board is reseting in current mode")
+        self.reset(self.qspi_cfg)
```

Comment:
Hmm. The supported boot modes will likely differ often from board to board, so hard-coding them in the BootCfgProtocol doesn't seem very useful. Perhaps only use a single method with a string option? Perhaps upload the code for your second driver somewhere (even if just in a https://gist.github.com/) so we could get a better impression.

The BootCfgProtocol class could still be added when adding the second implementation.

Reply:
Ok, so for bcu driver I will remove the BootCfgProtocol - indeed supported boot modes might differ between boards ; using a single method with string parameter seems a better option (I was thinking at it back then when I implemented the driver but I can't remember why I didn't use it). 
I will try to upload the second driver as you suggested, but it might take some time.